File: app/src/main/java/com/lightsoft/parceler/model/Phone.java
package com.lightsoft.parceler.model;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@SuppressWarnings("WeakerAccess")
@Parcel(Parcel.Serialization.BEAN)
public class Phone extends Device {
    private String phoneNumber;

    @ParcelConstructor
    public Phone(String manufacturer, String model, String deviceId, String phoneNumber) {
        super(manufacturer, model, deviceId);
        this.phoneNumber = phoneNumber;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        if (!super.equals(o))
            return false;

        Phone phone = (Phone) o;

        return phoneNumber != null ? phoneNumber
                .equals(phone.phoneNumber) : phone.phoneNumber == null;
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + (phoneNumber != null ? phoneNumber.hashCode() : 0);
        return result;
    }
}

File: kotlinparceler/src/main/java/com/lightsoft/kotlinparceler/model/Phone.kt
package com.lightsoft.kotlinparceler.model

import org.parceler.Parcel
import org.parceler.ParcelConstructor
import org.parceler.ParcelProperty

@Parcel(Parcel.Serialization.BEAN)
class Phone
@ParcelConstructor
constructor(@ParcelProperty("manufacturer") manufacturer: String?,
            @ParcelProperty("model") model: String?,
            @ParcelProperty("deviceId") deviceId: String?,
            @ParcelProperty("phoneNumber") var phoneNumber: String?) : Device(manufacturer, model, deviceId)
{
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false
        if (!super.equals(other)) return false

        other as Phone

        if (phoneNumber != other.phoneNumber) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + (phoneNumber?.hashCode() ?: 0)
        return result
    }
}